js/framework/application.js

```diff
@@ -445,15 +456,25 @@ var Application = Class.inherit({
     _showViewImpl: abstract,
 
     _obtainViewLink: function(viewInfo) {
-        var key = viewInfo.key;
+        var key = viewInfo.key,
+            hash = this._viewLinksHash[key];
 
-        if(!this._viewLinksHash[key]) {
+        var createViewLinkHash = function() {
```

Comment:
This function looks strange to me. Can't we remove it and use the following code below:

```
        if(!viewLink) {
             this._viewLinksHash[key] = {
                 viewInfo: viewInfo,
                 linkCount: 1
             };
        } else {
            if(viewLink.viewInfo !== viewInfo) {
                this._removedViewInfos.push(viewLink.viewInfo);
                viewLink.viewInfo = viewInfo;
                // should viewLink.linkCount already be 1 here ???
            } else {
                viewLink.linkCount++;
            }
         }
```